encrypted-config-value/src/main/java/com/palantir/config/crypto/EncryptedValue.java

```diff
@@ -18,21 +18,32 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 
+import com.google.common.io.BaseEncoding;
 import com.palantir.config.crypto.algorithm.Algorithm;
 import com.palantir.config.crypto.algorithm.Algorithms;
 import java.io.IOException;
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
 
 @Value.Immutable
 public abstract class EncryptedValue {
 
     private static final String PREFIX = "enc:";
 
-    public abstract String encryptedValue();
+    @Nullable
+    public abstract String getAlgorithm();
```

Comment:
We are - encrypted values produced after this change look like `${enc:AES:3Jw0mA0TbPOqpqskhMx2hqC5cdfTWG/nPi0ut36cPkPax5ZL1uWrVJOVhTZ/LfbsO1H6ZLVXP0b/Ng3quYvvYnK2Sx9+KZM=}`. The `AES` bit in there is parsed and stuck in the algorithm field of `EncryptedValue`. The issue is that we still need to support the old form (e.g. `${enc:QjR4AHIYoIzvjEHf53XETM3QYnCl1mgFYC51Q7x4ebwM+h3PHVqSt/1un/+KvpJ2mZfMH0tifu+htRVxEPyXmt88lyKB83NpesNJEoLFLL+wBWCkppaLRuc/1w==}`) which is missing the algorithm specifier. Upstream deployment infrastructure stuff will still be producing this old format for some amount of time since it seems unnecessary to break deserialization back compat and force the entire ecosystem to upgrade at once. When we do decide to make that break, we can remove the `Optional` here.